openmeetings-db/src/main/java/org/apache/openmeetings/db/entity/file/BaseFileItem.java

```diff
@@ -296,7 +297,31 @@ public String getExternalType() {
 	public void setExternalType(String externalType) {
 		this.externalType = externalType;
 	}
-
+	
+	/**
+	 * Generates a link to download. As opposed to display it on the Whiteboard.
+	 * 
+	 * @param ext
+	 * @return
+	 */
+	public final File getDownloadFile(String ext) {
+		// passing in null for PRESENTATION won't return the original
+		if (ext == null && getType() != null && getType().equals(Type.PRESENTATION)) {
+			// could by either a Doc/XLS/PPT or just a PDF
+			return getFile(FilenameUtils.getExtension(getName()));
```

Comment:
`getName()` is the name of the file which is originally the same as the name of the file uploaded

BUT :))

1. OM automatitically detects file type based on contents i.e. the type will be PRESENTATION if you will rename `test..txt` to `test.avi`
2. user can change the name any time with double-click and edit

`OMFileHelper` contains method to get extension, so I would say we need to
1. use it
2. or replace with `FilenameUtils` in all places

Reply:
Yeah I didn't think about the user can change the name.
=> but where would you then find out what the original file name extension was in case of a document of type PPT/XLS/DOCX et cetera ?

Reply:
Due to above

The logic was following:
File folder for `IMAGE` contains
1. original image
2. ** image as `PNG` (will be the same if image uploaded was `PNG`)
So to get Original we should list files and filter `PNG` :))

File folder for `PRESENTATION` contains
1. original documment
2. `PDF` (if document wasn't `PDF`)
3. list of pages as `PNG`
So to get Original we should list files and filter `PNG` and `PDF` :))

This logic can be tweaked to return `PNG` for image if there is only `PNG` available
Similar rule for `PRESENTATION`

Reply:
well the behaviour as is is as follows: If you have a presentation and you download original it returns the first slide of the document converted to PNG.
So you click "download original" and you get prompted with a .png file.
Just try it.

Reply:
I just have tried:

1. Upload 4 page `test.odt`
2. Download original
Result: `--hash--.odt` was downloaded

What were your steps?